Source/Scene/GltfLoader.js

```diff
@@ -979,8 +986,23 @@ function loadInstancedAttribute(
     loader._loadAttributesFor2D &&
     !frameState.scene3DOnly;
 
-  const loadBuffer = !loadAsTypedArrayPacked;
-  loadAsTypedArrayPacked = loadAsTypedArrayPacked || loadFor2D;
+  // In addition to the loader options, load the attributes as typed arrays if:
+  // - the instances have rotations, so that instance matrices are computed on the CPU.
+  //   This avoids the expensive quaternion -> rotation matrix conversion in the shader.
+  // - the translation accessor does not have a min and max, so the values can be used
+  //   for computing an accurate bounding volume.
+  // - the attributes contain feature IDs, in order to add the instance's feature ID
+  //   to the pick object.
+  // - translations are required for 2D
+  // - GPU instancing is not supported.
+  let loadTypedArray =
+    loader._loadAttributesAsTypedArray ||
+    ((hasRotation || !hasTranslationMinMax) && isTransformAttribute) ||
+    modelSemantic === InstanceAttributeSemantic.FEATURE_ID ||
+    !frameState.context.instancedArrays;
+
+  const loadBuffer = !loadTypedArray;
+  loadTypedArray = loadTypedArray || loadFor2D;
```

Comment:
I tried to put all the load-as-typed-array logic in one place. Previously it was spread across this function and the calling code.